test/box-unbox.js

```diff
@@ -0,0 +1,286 @@
+'use strict'
```

Comment:
this file used to be called `end-to-end.js`. I thought the name was misleading.
It's been renamed, and it's ~85% the same. I had to change some parts, and refactored a little as I went to be able to parse what the tests were doing

Importantly I've added tests around : 
- `addBoxer`
- `addUnboxer` (both signatures, and new async `init`)